ash/src/vk/macros.rs

```diff
@@ -1,103 +1,95 @@
-                self & other != $name::empty()
+            pub fn intersects(self, other: Self) -> bool {
+                self & other != Self::empty()
             }
             #[doc = r" Returns whether `other` is a subset of `self`"]
             #[inline]
-            pub fn contains(self, other: $name) -> bool {
+            pub fn contains(self, other: Self) -> bool {
                 self & other == other
             }
         }
         impl ::std::ops::BitOr for $name {
-            type Output = $name;
+            type Output = Self;
             #[inline]
-            fn bitor(self, rhs: $name) -> $name {
-                $name(self.0 | rhs.0)
+            fn bitor(self, rhs: Self) -> Self {
+                Self(self.0 | rhs.0)
             }
         }
         impl ::std::ops::BitOrAssign for $name {
             #[inline]
-            fn bitor_assign(&mut self, rhs: $name) {
+            fn bitor_assign(&mut self, rhs: Self) {
                 *self = *self | rhs
             }
         }
         impl ::std::ops::BitAnd for $name {
-            type Output = $name;
+            type Output = Self;
             #[inline]
-            fn bitand(self, rhs: $name) -> $name {
-                $name(self.0 & rhs.0)
+            fn bitand(self, rhs: Self) -> Self {
+                Self(self.0 & rhs.0)
             }
         }
         impl ::std::ops::BitAndAssign for $name {
             #[inline]
-            fn bitand_assign(&mut self, rhs: $name) {
+            fn bitand_assign(&mut self, rhs: Self) {
                 *self = *self & rhs
             }
         }
         impl ::std::ops::BitXor for $name {
-            type Output = $name;
+            type Output = Self;
             #[inline]
-            fn bitxor(self, rhs: $name) -> $name {
-                $name(self.0 ^ rhs.0)
+            fn bitxor(self, rhs: Self) -> Self {
+                Self(self.0 ^ rhs.0)
             }
         }
         impl ::std::ops::BitXorAssign for $name {
             #[inline]
-            fn bitxor_assign(&mut self, rhs: $name) {
+            fn bitxor_assign(&mut self, rhs: Self) {
                 *self = *self ^ rhs
             }
         }
         impl ::std::ops::Sub for $name {
-            type Output = $name;
+            type Output = Self;
             #[inline]
-            fn sub(self, rhs: $name) -> $name {
+            fn sub(self, rhs: Self) -> Self {
                 self & !rhs
             }
         }
         impl ::std::ops::SubAssign for $name {
             #[inline]
-            fn sub_assign(&mut self, rhs: $name) {
+            fn sub_assign(&mut self, rhs: Self) {
                 *self = *self - rhs
             }
         }
         impl ::std::ops::Not for $name {
-            type Output = $name;
+            type Output = Self;
             #[inline]
-            fn not(self) -> $name {
-                self ^ $name::all()
+            fn not(self) -> Self {
+                Self(!self.0)
```

Comment:
The best we can do here is fully negate the bitflags since we never really know what "all" bitflags are - even if we decide to keep and improve `fn all()` in the end.

(Sidenote: should I create a separate PR for te `$name` -> `Self` conversion so that these details pop out better?)

Reply:
> The best we can do here is fully negate the bitflags

Seems reasonable to me. I think the only case where you'd be negating bitflags is for purposes of constructing masks or something, in which case this is exactly the desired behavior.

> should I create a separate PR for te $name -> Self conversion

This would make it easier to carefully review, yeah.